File: backend/src/modules/maestros/maestros.controller.js
// import maestrosService from './maestros.service.js';

const maestrosController = {};
const JAVA_API = 'http://localhost:8080/api/java/maestros';

maestrosController.obtenerPerfil = async (req, res) => {
    try {
        const idUsuario = req.params.id;
        const response = await fetch(`${JAVA_API}/${idUsuario}/perfil`);
        const data = await response.json();
        
        if (!response.ok) {
            return res.status(response.status).json(data);
        }

        res.status(200).json(data);
    } catch (error) {
        console.error('Error al obtener perfil del profesor:', error);
        res.status(500).json({ success: false, message: 'Error interno del servidor' });
    }
};

export default maestrosController;

File: java-backend/src/main/java/com/example/mathexpedition/controllers/MaestrosController.java
package com.example.mathexpedition.controllers;

import com.example.mathexpedition.models.Maestro;
import com.example.mathexpedition.repository.MaestroRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/maestros")
@CrossOrigin(origins = "*")
public class MaestrosController {

    @Autowired
    private MaestroRepository repoMaestros;

    @GetMapping("/{id}/perfil")
    public ResponseEntity<Map<String, Object>> obtenerPerfil(@PathVariable String id) {
        Maestro maestro = repoMaestros.obtenerPorId(id);

        if (maestro == null) {
            Map<String, Object> error = new HashMap<>();
            error.put("success", false);
            error.put("message", "Perfil de profesor no encontrado");
            return ResponseEntity.status(404).body(error);
        }

        Map<String, Object> result = new HashMap<>();
        result.put("idUsuario", maestro.get_idUsuario() != null ? maestro.get_idUsuario() : maestro.getIdUsuario());
        result.put("name", maestro.getName());
        result.put("email", maestro.getEmail());

        Map<String, Object> success = new HashMap<>();
        success.put("success", true);
        success.put("perfil", result);
        return ResponseEntity.ok(success);
    }
}
